Add documentation for the new Symfony kernel browser driver (#1)
blackfireio/documentation: php/integrations/behat.rst

- Behat 3
- ``friends-of-behat/mink-extension`` >= 2.0
- ``friends-of-behat/mink-browserkit-driver`` >= 1.5

> - ``friends-of-behat/symfony-extension`` >= 2.0 (Optional, only required if you want to use the Symfony driver)

Reviewer: ```suggestion
- ``friends-of-behat/symfony-extension`` >= 2.0 (Optional. Required only with the Symfony driver)
```

.. note::

>     Note that the original ``behat/mink-extension``, ``behat/symfony2-extension`` and ``minkphp/mink-browserkit-driver``

Reviewer 2: ```suggestion
    Note that the original ``behat/mink-extension``, ``behat/symfony2-extension``, and ``minkphp/mink-browserkit-driver``
```

    composer require blackfire/php-sdk friends-of-behat/mink-extension friends-of-behat/mink-browserkit-driver

> If you want to use the Symfony driver, also install the Symfony extension:

Reviewer: ```suggestion
Install the Behat Symfony extension If you'd like to use the Symfony driver:
```

Blackfire driver. Every step in your Behat scenario is going to be profiled
unless you tell it otherwise from a feature context.

> If you use the above configuration, tests will be run using the Browserkit driver which performs actual requests
> against your code. If you are using Symfony, you may prefer to use the kernel browser driver, which does not perform
> actual requests, only simulates them by directly calling the Symfony kernel and which is much faster. In this case,
> change the above configuration to:

Reviewer: ```suggestion
Optionally, if you are using :doc:`Symfony </php/integrations/symfony/index>`,
you may use the kernel browser driver. This driver does not perform actual
requests. It only simulates them by calling the Symfony kernel directly, which is
faster. In this case, change the above configuration to:
```

        }
    }

> .. note::
> 
>     If you use the Symfony driver, since no actual requests are made, Blackfire will interpret these profiles as coming
>     from a command, not an HTTP request, so it will apply the assertions that you have defined for your commands,
>     not your requests

Reviewer: ```suggestion
.. warning::

    If you use the Symfony driver, Blackfire will interpret these profiles as
    coming from a command, not an HTTP request, since no actual requests are made.
    The assertions defined for your commands will be evaluated, and not those
    for your HTTP requests.
```

Reviewer: @carlos-granados, this said, I don't like we have to end up in this situation with Symfony driver.

For Laravel Octane (swoole-based), we were in a similar situation, but for the monitoring. Using this snippet helped flag the traces as HTTP requests.
https://github.com/blackfireio/php-sdk/blob/master/src/Blackfire/Bridge/Laravel/Listeners/OctaneRequestIntrumentationStart.php#L31C8-L36C63

@romainneutron, do you think it could also work for the profiles triggered using the Symfony Driver?

Author: Let me try it, I will let you know if that works

Author: @thomasdiluccio Unfortunately it does not seem to work. I tried both calling the `setAttribute` static function as in the Laravel Octane code and calling the `setAttribute` function on the $probe object that is created when the request is started and it still understands the profiles as coming from a command instead of an HTTP request :-(

Reviewer: @romainneutron Do you see a way to force the profile to be interpreted as an HTTP call?

Reviewer 3: unfortunately no

>     The assertions defined for your commands will be evaluated, and not those
>     for your HTTP requests.

Reviewer 2: ```suggestion
    Blackfire will therefore evaluate the assertions defined for your commands instead of those defined for your HTTP requests.
```
